components/encoder/single/single_encoder.go

```diff
@@ -175,31 +168,39 @@ func (e *Encoder) Reconfigure(
 	atomic.StoreInt64(&e.position, 0)
 	e.mu.Unlock()
 
-	e.Start(ctx, board)
+	e.start(ctx, board)
 
 	return nil
 }
 
-// Start starts the Encoder background thread.
-func (e *Encoder) Start(ctx context.Context, b board.Board) {
+// start starts the Encoder background thread.
+func (e *Encoder) start(ctx context.Context, b board.Board) {
+	if e.workers != nil {
```

Comment:
This should never occur, but I couldn't think of a way to assert that at compile-time. I'm willing to remove this if statement, if you want.

Reply:
When I wrote this orignially, `Start` was a public function that anyone could call and mess things up. Now that it's private, we don't need to worry about that any more, and this can all be removed.

Reply:
...well, it's good defensive programming in case someone makes changes in the future. Without this, we need the invariant that `e.workers` is either `nil` or already stopped before anyone calls `start()`, and with this, although that's still how to think about things, it's "okay" for someone to violate that invariant. Maybe keep this as-is. I can go either way.